crates/solidity/testing/utils/src/fetch.rs

```diff
@@ -24,13 +27,45 @@ pub fn fetch(address: &str, base_path: &Path) -> Result<()> {
 
     // Try with exponential backoff
     let mut tries = 0;
-    while body.is_err() && tries < 6 {
+    while !body
```

Comment:
The proposed code will fail on invalid JSON without retrying. I suggest:

- using `loop` and `break value` to `get()` only once
- check headers/decode the JSON inside the retry loop
- print errors on every attempt, instead of only the last one, for us to check if the error varies between retries.


```rust
let mut i = 0;
let project = loop {
  i += 1;
  match try_fetch_project(&url) {
    Ok(project) => break project,
    Err(err) => {
      println!("Error fetching JSON on attempt {i} of {MAX_RETRIES}: {err}");
      if i <= MAX_RETRIES {
        sleep(...);
      } else {
        panic!("Giving up.");
      }
    }
  }
}

fn try_fetch_project(url) {
  // fetch response, check status and headers, and decode JSON in the same retry
}
```